vagrant/environments/pos/config.yaml

```diff
@@ -0,0 +1,110 @@
+---
+# Default settings for all vms (they can be overridden on each node)
+vm:
+  memory: 1024                 # MB or RAM to assign
+  cpu: 1                       # Number of vCPU to assign to the VM
+  box: centos7                 # Box used for the VM, from the boxes list below
+  puppet_apply: false          # Run puppet apply on the local control-repo during provisioning
+  puppet_agent: true           # Run puppet agent during provisioning
+  facter_external_facts: false # Create external facts in facts.d/$fact.txt. Note 1
+  facter_trusted_facts: true   # Create csr_attributes.yaml. Note 1
+
+# Note 1: Some facts are used in default hiera.yaml and needed for
+# nodes classification. So it's better to set them, once.
+# If you enable both external and trusted facts and keep the
+# default manifests/site.pp you'll have a variable declaration error 
+
+# A local network is created among the VM. Here is configured.
+network:
+  range: 10.42.46.0/24   # Network address and mask to use 
+  ip_start_offset: 101   # Starting ip in the network for automatic assignement
+  domain: pos.psick.io    # Name of DNS domain for the created machines
+
+# Puppet related settings
+puppet:
+  version: latest             # Version to use for OSS
+  install_oss: true          # If to install Puppet OSS agent on the VMS
+  install_pe: false            # If to install Puppet Enterprise agent on the VMS
+  env: devel                  # Setting for the env fact (may be used in hiera.yaml)
+  zone: pos                    # Setting for the zone fact (may be used in hiera.yaml)
+  datacenter: vagrant         # Setting for the datacenter fact (may be used in hiera.yaml)
+  application: default        # Setting for the application fact (may be used in hiera.yaml)
+  master_vm:  puppet.pos.psick.io      # Name of the VM which play as Puppet server for the others
+  master_fqdn: 'puppet.pos.psick.io'   # FQDN of the Puppet server to use with puppet agent
+  link_controlrepo: true      # Add a link for a Puppet environment to the development control-repo 
+  environment: production     # Puppet environment to link to local control-repo
```

Comment:
if you set this to host the setup should work without changing vagrant/bin/papply.sh

this unusual host environment is used from vagrant envs where we have a puppetserver